### DISCUSSION TITLE
Screen size larger than 256 pixels?

### DISCUSSION THREAD

**Discussion by JimDrewGH** [Roger-random/ESP_8_BIT_composite#14] [Ideas]
Is it possible to add the 320 pixel wide mode to this library?

**Comment by Roger-random** [comment]
320 pixel wide mode was not available in the ESP_8_BIT project from which I extracted the composite signal generation code. So adding 320 pixel wide mode will take some work and not trivial.

But you didn't ask for **trivial**, you asked if it is **possible**, and that I'm not sure until I have a better understanding of the original code.

If anyone figures it out, I would certainly review the pull request to see if makes sense to incorporate into this Arduino library.

**Reply by JimDrewGH** [reply]
I believe that the ESPFlix project (which is by Rossumur and uses the same code base) uses a 344 wide pixel display.  Many of the various retro machines and arcade games (C64 and DOOM for examples) used a 320 wide display.  It would be very handy to be able to have a 320x200 or 320x240 (although 212 is the max allowed for NTSC TV displays) output with 256 colors from a palette of valid YUV values.

**Reply by Roger-random** [reply]
If someone just wants to hack around, 320 / 256 = 1.25. Dropping every fifth pixel wouldn't look very good, but it would be a starting point to render 256 out of 320 pixels before getting everything else figured out.

I looked at ESPFlix and it's not immediately obvious to me how to adapt its wider pixel display to this project. For one thing the color signal generation is different, as ESPFlix is focused on YUV colorspace. ESPFlix also has more limited vertical resolution, due to optimization for letterboxed aspect ratios. (Or I'm misunderstanding the code, which is certainly possible.)

**Reply by Roger-random** [reply]
Update: Horizontal resolution wider than 320 pixels were available on the Atari emulator code path, which I never got running on my own ESP32 but @marciot did. See discussion #20 for more details on the exploration.
